lib/spack/spack/test/web.py

```diff
@@ -436,7 +437,7 @@ def test_ssl_curl_cert_file(cert_exists, tmpdir, ssl_scrubbed_env, mutable_confi
         spack.config.set("config:ssl_certs", mock_cert)
         if cert_exists:
             open(mock_cert, "w").close()
-            assert os.path.isfile(mock_cert)
+            assert Path(mock_cert).is_file()
```

Comment:
`tmpdir.join(...` will produce a pathlib object, instead of casting to pathlib here, cast `mock_cert` to string in the call to `spack.config.set` on linke 437.